### PR TITLE
Add eslint infrastructure

### PR DESCRIPTION
**Pull request by jannyHou**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "node": ">=0.10.0"
   },
   "scripts": {
-    "pretest": "jshint .",
+    "posttest": "eslint .",
```
Comment (superkhau) [inline package.json]: move this line under "test"

**Comment by superkhau** [comment]
One minor comment, otherwise LGTM

File: package.json (hunk 1 above)
Comment (eddiemonge) [inline package.json]: why is it being moved to posttest?

**Comment by jannyHou** [comment]
The test cases are broken now so I prefer to finish this pr after it's fixed.

**Comment by eddiemonge** [comment]
closing in favor of #189 which includes this change and also the changes to make eslint happy

File: package.json (hunk 1 above)
Comment (bajtos) [inline package.json]: When doing red-green-refactor cycle, we don't want to wait for linter every time we run the test suite. Another way of looking at this: first get the implementation right, only then worry about the (coding) style.

**Pull request closed without merging**